api/features/views.py

```diff
@@ -154,8 +154,32 @@ def get_queryset(self):
         )
         queryset = queryset.order_by(sort)
 
+        if environment_id:
+            page = self.paginate_queryset(queryset)
+
+            if not getattr(self, "environment", None):
+                self.environment = Environment.objects.get(id=environment_id)
```

Comment:
I'm not sure under which scenarios here we do / don't have the environment already on the instance?

Reply:
I believe I did this to avoid loading the environment twice with a separate PR which similarly needed the environment. But on second look I think the environment will always be loaded in this codepath first, so I'll update this to just set the environment without checking.